poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -91,16 +97,19 @@ impl Extend<Heartbeat> for Heartbeats {
             if heartbeat.validity != proto::HeartbeatValidity::Valid {
                 continue;
             }
-            self.heartbeats.insert(
-                HeartbeatKey {
+            let entry = self
+                .heartbeats
+                .entry(HeartbeatKey {
                     hotspot_key: heartbeat.hotspot_key,
                     cbsd_id: heartbeat.cbsd_id,
-                },
-                HeartbeatValue {
-                    reward_weight: heartbeat.reward_weight,
-                    timestamp: heartbeat.timestamp,
-                },
-            );
+                })
+                .or_default();
+            entry.reward_weight = heartbeat.reward_weight;
+            if entry.timestamps.is_empty()
+                || entry.timestamps[0].hour() != heartbeat.timestamp.hour()
+            {
+                entry.timestamps.insert(0, heartbeat.timestamp);
```

Comment:
should we be concerned with any performance implications of having to shift all the elements of the list right here vs. pushing newer elements onto the vector and then grabbing the last index instead of the first?

Reply:
This is only used when we do the generate cli so I don't think so